### PR TITLE
Add `warnonce` function to trigger a warning only once

### PR DESCRIPTION
**Pull request by oerc0122**
Related to #1609 

Does not use this function anywhere, but provides function for later PRs

### CODE REVIEW

File: _test/test_utilities/test_warnonce.m [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        function obj = test_warnonce(~)
+            obj@TestCase('test_warnonce');
+        end
+
+        function test_warnonce_warns_once_defaults(~)
+            out = evalc("warnonce('HORACE:panic:panic', 'Panic at the disco')");
+            msg = lastwarn;
+
+            assertEqual(msg, 'Panic at the disco');
+            assertTrue(contains(out, 'Panic at the disco'))
```
Comment (abuts) [inline _test/test_utilities/test_warnonce.m:13]: The test will fail if run on the same machine twice. 

This is pain for maintenance. 
Reset mechanism should be added for testing purposes.

**Review by abuts — CHANGES_REQUESTED** [review]
idea ok, but tests would fail if run on the same machine twice. This should be avoided
simple reset 
(e.g. call with:
 `warnonce('-reset')` should be implemented for testing purposes. Or `warnonce('')` like `lastwarn` even better

File: _test/test_utilities/test_warnonce.m (hunk 1 above)
Comment (oerc0122) [reply _test/test_utilities/test_warnonce.m:23]: Double check you're happy with the new mechanism and we can start putting this in places where it's used.

**Review by cmarooney-stfc — APPROVED** [review]
on the basis this is infrastructure only, needs propagating

**Pull request merged**